src/components/form/SubmitButton.tsx

```diff
@@ -8,7 +8,8 @@ import { ProgressButton, ProgressButtonProps } from '../ProgressButton';
 export type SubmitButtonProps = {
   /** Whether to show progress while submitting */
   spinner?: boolean;
-} & Omit<ProgressButtonProps, 'type' | 'disabled' | 'progress'>;
+  disabled?: boolean;
```

Comment:
SubmitButton should be disabled based on whether the form state is valid or submitting which happens internally. Was there a reason to add it here?

Reply:
I removed this and made it so the dialog is disabled if the data hasn't been loaded yet.